src/ripple/app/main/Main.cpp

```diff
@@ -470,6 +471,12 @@ int run (int argc, char** argv)
                 "Debug", beast::severities::kTrace));
         }
 
+        std::set_terminate ([]()
+        {
+            LogThrow ("Unhandled exception");
+            std::abort ();
+        });
+
```

Comment:
On OS X this gives an inferior stack dump compared to what already happens.  With this change I get no output at the terminal with an unhandled exception, and the stack dump summary is "abort() called".  Without the change (and with the simulated unhandled exception I forced), I get:

libc++abi.dylib: terminating with uncaught exception of type ripple::SHAMapMissingNode: SHAMapMissingNode

both on the terminal, and in the stack dump.  This is both the type of the exception, and its what() message.

Guarding with:
````
#ifndef __APPLE__
````
is one possible solution.